### PR TITLE
Destructible Walls (And Mineral Walls)

### PR DESCRIPTION
**Pull request by Gristlebee**
### Wall Stuff and Behavior
- Moves the damage behavior from concrete walls to closed turfs.
- Basic walls can be repaired with a welder.
- Basic walls are still deconstructed with a welder, but it now starts a repeatable do after that reduces the wall's integrity more efficiently than simply attacking. 
- R-walls integrity is tied to their deconstruction state. Damaging one enough will move it down a state, and can be repaired by following the construction steps. They can still be deconstructed the same as before, or with a plasmacutter.
### Plasmacutters
- Plasma cutters are now more of an engineering tool, and acts as an "upgrade" to the welder and deconstructs walls faster, and can repair prosthetic limbs. They can still mine, but lose the additional mining range and has a chance to ruin ores. 
- Plasmacutters now fit in toolbelts.
- Moves plasmacutters and advanced plasmacutters from mining to tech plasma manipulation and adv plasma manipulation tech respectively.
- Adds plasmacutters to the Mudskipper, Riggs, Talos, Gecko, Heron, Osprey, Ranger, Hyena, Komodo, Shetland and Twinkleshine
- Adds plasmacutters to cargo for 1250 credits

### Misc Code stuff
- Cleans up crate/closet deconstruction. It now checks for tool_behavior rather than a specific tool and non-help intent.
- Renames concrete_damage.dmi to wall_damage.dmi
- Moves indestructible walls into it's own file
- bitflags for bonus damage on walls

## Why It's Good For The Game

Rylie asked me to make this.

Currently, it's much easier to go through a wall than a door if you want to force entry as it's as simple as having a welder or a jackhammer to take off the wall plating vs having to take the time hack an airlock, which probably isnt great. Also gives some counterplay if you're on the defending side since you have the opportunity to repair compromised walls or go around and attack the breacher.

This would add more possible interactions and play around with walls since you no longer need these two specific tools to get through a wall. If you say, have enough ammo you could theoretically just shoot your way in, and they're no longer indestructible pieces of cover.

## Changelog

:cl:
add: Welders/Plasmacutters now deal damage to wall integrity to decon
add: Plasmacutters are now engineering tools and fit in toolbelts. They can damage ores into slag if used to mine. They are now researched with plasma technology and advanced plasma technology. 
add: Plasmacutters to cargo for 1250
add: Adds plasmacutters to the Mudskipper, Riggs, Talos, Gecko, Heron, Osprey, Ranger, Hyena, Komodo, Shetland and Twinkleshine
del: Plasmacutters extra mining range
fix: Plasmacutters can repair prosthetic limbs like welders
code: Moves damage behavior of concrete walls to closed turfs. Basic walls are now repaired with a welder.
code: R-walls d-state tied to their integrity. R-walls can be deconstructed with plasmacutters.
code: Crate decon checks for tool behavior rather than a istype check

/:cl:

### CODE REVIEW

**Review by thgvr — DISMISSED** [review]
This needs some form of visual and audio feedback on damaged walls

File: _maps/shuttles/nanotrasen/nanotrasen_gecko.dmm [hunk 1]
```diff
@@ -1548,6 +1548,7 @@
 /obj/item/clothing/glasses/meson/engine/tray,
 /obj/item/tank/internals/emergency_oxygen/engi,
 /obj/item/radio/intercom/directional/north,
+/obj/item/gun/energy/plasmacutter,
```
Comment (Sadhorizon) [inline _maps/shuttles/nanotrasen/nanotrasen_gecko.dmm]: Shouldn't those be in salvage tech lockers instead? Salvaging's in the name!

Comment (Gristlebee) [reply _maps/shuttles/nanotrasen/nanotrasen_gecko.dmm]: my brain told me engineering tool so into the engi locker, but yeah this makes sense. It's been moved.

File: code/modules/cargo/packs/tools.dm [hunk 2]
```diff
@@ -74,6 +74,12 @@
 	contains = list(/obj/item/pickaxe/drill/jackhammer)
 	crate_name = "jackhammer crate"
 
+/datum/supply_pack/tools/jackhammer
```
Comment (Thera-Pissed) [inline code/modules/cargo/packs/tools.dm]: overwrites the other jackhammer crate, probably meant to be tools/plasmacutter or similar.

Comment (Gristlebee) [reply code/modules/cargo/packs/tools.dm]: yeeeep. That it is. Thanks for catching that

**Comment by Gristlebee** [comment]
Quashed a last few bugs, and the sounds are now actually working as intended. The PR should be good for review.

**Comment by Anticept** [comment]
There is an unintentional side effect with ruins that have emitters, they will eventually punch through walls and hit things on the map. Ruins do need a type of indestructible wall and hopefully one still exists so it can be used to block those emitters.

**Comment by Gristlebee** [comment]
> There is an unintentional side effect with ruins that have emitters, they will eventually punch through walls and hit things on the map. Ruins do need a type of indestructible wall and hopefully one still exists so it can be used to block those emitters.

yeah there's an indestructible wall subtype that can be used. If you already know which ruins specifically have this issue so I can change em, that'd be a great help

**Comment by firebudgy** [comment]
Harmfactory has two emitters shooting regular rwalls

**Comment by Gristlebee** [comment]
Harmfactory's gotten, and it seems like thats the only ruin with this issue

**Comment by thgvr** [comment]
@tmtmtl30 please review

**Review by FalloutFalcon — APPROVED** [review]
Looks good code wise, haven't been able to see it in game.

**Comment by rye-rice** [comment]
if tmt doesnt review this by tommorow i will take that as an approval and merge

**Comment by rye-rice** [comment]
okay nvm apparently tmt already reviewed it in a voice call

**Pull request merged**